File: schemas/recipe-cooking/mongodb/units.js
// units: Measurement units for ingredient quantities.
// See README.md for full design rationale.

const mongoose = require("mongoose");

const unitSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, unique: true },
    abbreviation: { type: String, default: null },
    system: {
      type: String,
      enum: ["metric", "imperial", "universal"],
      default: null,
    },
  },
  {
    timestamps: false,
  }
);

module.exports = mongoose.model("Unit", unitSchema);

File: schemas/recipe-cooking/spacetimedb/units.rs
// units: Measurement units for ingredient quantities.
// See README.md for full design rationale.

#[derive(SpacetimeType, Clone)]
pub enum UnitSystem {
    Metric,    // type: String
    Imperial,
    Universal,
}

#[spacetimedb::table(name = units, public)]
pub struct Unit {
    #[primary_key]
    pub id: String, // UUID

    #[unique]
    pub name: String,

    pub abbreviation: Option<String>,
    pub system: Option<UnitSystem>,
}
